airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py

```diff
@@ -664,6 +666,46 @@ def _flat_associations(self, records: Iterable[MutableMapping]) -> Iterable[Muta
             yield record
 
 
+class SemiIncrementalStream(Stream, IncrementalMixin):
+    _cursor_value = 1
+
+    @property
+    def cursor_field(self) -> Union[str, List[str]]:
+        return self.updated_at_field
+
+    @property
+    def state(self) -> Mapping[str, Any]:
+        return {self.cursor_field: self._cursor_value}
+
+    @state.setter
+    def state(self, value: Mapping[str, Any]):
+        self._cursor_value = value[self.cursor_field]
+
+    def filter_by_state(self, stream_state: Mapping[str, Any] = None, record: Mapping[str, Any] = None) -> bool:
+        int_field_type = "integer" in self.get_json_schema().get("properties").get(self.cursor_field).get("type")
+        record_value, datetime_format = (
+            (pendulum.parse(record.get(self.cursor_field)), "YYYY-MM-DDTHH:mm:ss.SSSSSSZ")
+            if isinstance(record.get(self.cursor_field), str)
+            else (pendulum.from_format(str(record.get(self.cursor_field)), "x"), "x")
```

Comment:
what is `x`?  also can you add a code comment about why we have this `if` statement?

Reply:
docsting added

Reply:
this is a smaller comment but fwiw this should be done using inversion of control i.e: the super class should not know what the subclasses do. Instead the subclasses should have a way to provide the datetime object given their knowledge of their own date format.